src/Compilers/CSharp/Portable/FlowAnalysis/AbstractFlowPass.cs

```diff
@@ -959,7 +959,16 @@ public override BoundNode VisitIsPatternExpression(BoundIsPatternExpression node
         {
             Debug.Assert(!IsConditionalState);
             VisitRvalue(node.Expression);
-            VisitPattern(node.Pattern);
+
+            var pattern = node.Pattern;
+            bool negated = false;
+            while (pattern is BoundNegatedPattern n)
```

Comment:
>while (pattern is BoundNegatedPattern n) [](start = 12, length = 40)

might we encounter parentheses in this stack of `not`s? Or do parens patterns simply get omitted from bound nodes?

Reply:
There is no bound pattern for the parens. They disappear in the bound nodes.

---
In reply to: [430768943](https://github.com/dotnet/roslyn/pull/44513#discussion_r430768943) [](ancestors = 430768943)